### ISSUE TITLE
optionally reload page on consent state change.

### ISSUE THREAD

**Issue by jpettitt** [ampproject/amphtml#15406]
This is very much is the "nice to have" but not essential category.

If a user changes a previous consent it would be nice to have an option to reload the page so the user can see the impact of their action.

Related to #13716

**Comment by rolebi** [comment]
On our pure HTML pages, where we sometime have 50% of our users refusing cookies, we've put placeholders in place of third party contents that may load cookies or track our users (Youtube, Pinterest, Instagram, ....). Those placeholders kindly request the users to update their cookie policy settings if they want to access the underlying content. On click, we open the cookie policy preference center to allow the users to update their settings.

We have great result with this approach but sadly can't do the same on AMP pages, or more precisely, everything's doable up to the loading / reloading of the third party content on consent update, which is a shame : /

**Comment by levin-roski** [comment]
> On our pure HTML pages, where we sometime have 50% of our users refusing cookies, we've put placeholders in place of third party contents that may load cookies or track our users (Youtube, Pinterest, Instagram, ....). Those placeholders kindly request the users to update their cookie policy settings if they want to access the underlying content. On click, we open the cookie policy preference center to allow the users to update their settings.
> 
> We have great result with this approach but sadly can't do the same on AMP pages, or more precisely, everything's doable up to the loading / reloading of the third party content on consent update, which is a shame : /

@rolebi I really like the way you presented the flow for consent updates, but in AMP there is currently no way to show a placeholder/fallback when consent is not given yet or rejected.

And even after updating the consent it's kind of painful for the user to manually reload the page to see the blocked component. Most search engine visitors are gone by then.

This placeholder/fallback and auto-reload-option or component render on consent updates is not important for analytics or ads, which I found a big part of the community be interested in, but rather for rendered parts of the UI like amp-youtube, amp-soundcloud or even really needed third party components like Google ReCaptcha, which all set cookies at some place.

I didn't find any issue for a fallback/placeholder for UI elements blocked by consent feature, should I create a separate one?

**Comment by rolebi** [comment]
> I didn't find any issue for a fallback/placeholder for UI elements blocked by consent feature, should I create a separate one?

Well you kinda can display a placeholder and open the cookie banner on click (alas in a convoluted way), but either way, you won't be able to display the underlying content post consent update.

So yes, having this kind of workflow doable "natively" would greatly help !

**Comment by levin-roski** [comment]
Thanks @rolebi I tested this before I wrote my answer, but seemed like I tested it wrong.
Here is the example:
https://amp-rejected-consent-placeholder.glitch.me/

Nice to have this functionality already implemented.
Just needing a reload or render on consent update for this use case to work.
